### PR TITLE
[UI] New working draft of class statistics page

### PR DESCRIPTION
**Pull request by dvolpin29**
**Description**

- Working draft of a new class statistics page
- Created two new buttons in class-overview.html, to choose between two new Class Statistics pages: class-stats-v1.html, class-stats-v2.html
- Each page, displays student tiles with student information; name, quiz stats, program level progression, etc
-  Feedback from teachers will be gathered through interviews, and together with external research be used to create a final draft of the new class statistics page

This updated page should help teachers see a better overview of how each student is doing in their class.

**Discussion Number: 3447**

In the discussion board number 3447, Felienne has raised the issue of how the class statistics page looks. The proposal was to simplify it for the teachers so it is more useful.

**How to test**

* no testing needed at the moment

### CODE REVIEW

File: templates/class-stats-v2.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <button class="green-btn ml-auto" id="to_class_button"
+                onclick="window.location.href='/for-teachers/class/{{class_info.id}}'">
+          {{_('back_to_class')}}
+        </button>
+    </div>
+
+    <div class="flex pt-2">
+        <button class="green-btn ml-auto"
+                onclick="hedyApp.loadQuizChart({{ class_info.levels }}, {{ class_info.students }})">
+            Load quiz data
```
Comment (dvolpin29) [inline templates/class-stats-v2.html:16]: Make the quiz data load when the page is loading.

File: website/statistics.py [hunk 2]
```diff
@@ -362,3 +498,133 @@ def get_general_class_stats(students):
...
+    for program in programs:
+        if program['level'] == level:
+            program_runs_per_level[level - 1].append([program['name'], str(program.get("error"))])
+            if str(program.get("error")) == "True":
+                success_runs_per_level[level - 1].append(program['name'])
+            elif str(program.get("error")) == "False":
+                error_runs_per_level[level - 1].append(program['name'])
+
+
+def find_quizzes_per_level(quizzes_per_level, quizzes, level):
```
Comment (dvolpin29) [inline website/statistics.py:525]: https://github.com/hedyorg/hedy/pull/4207#discussion_r1224601894

**Comment by dvolpin29** [comment]
## This is the new Class Statistics page:
capture4

**Comment by Felienne** [comment]
Nice work @dvolpin29, it seems this PR represents the changes perfectly! Will take a look later today.

File: website/statistics.py [hunk 3]
```diff
@@ -362,3 +498,133 @@ def get_general_class_stats(students):
...
+    :param avg_quizzes_ran_per_level: list of average quizzes ran per level
+    :param quizzes_ran_per_level: list of lists containing the number of quizzes ran per level
+    :param level: current level
+    :return: updated list of average quizzes ran per level
+    """
+    if not quizzes_ran_per_level[level - 1]:
+        avg_quizzes_ran_per_level.append(0)
+    else:
+        quizzes_per_level = quizzes_ran_per_level[level - 1]
+        avg_quiz = sum(quizzes_per_level) / len(quizzes_per_level)
```
Comment (jpelay) [inline website/statistics.py:557]: All of the calculations look so much better now!! Thank you for changing them :D

**Comment by jpelay** [comment]
Thank you for taking into account my suggestions @dvolpin29! The calculations look nicer and cleaner.

**Comment by jpelay** [comment]
Hello @dvolpin29! We really like what you did in this PR and we appreciate all your hard work! However we're not going to merge this PR before  #4166 since both PRs are closely related and kind of solve the same issue. We'll keep this one open tho in case there's any work that you want to do here!

**Comment by Felienne** [comment]
I think we can close this now (it can always easily be reopened by @dvolpin29 if he wants to)

**Pull request closed without merging**